lib/pengine/utils.c

```diff
@@ -1009,24 +1051,11 @@ unpack_operation(pe_action_t * action, xmlNode * xml_obj, pe_resource_t * contai
 
     g_hash_table_remove(action->meta, "id");
 
-    // Normalize interval to milliseconds
-    field = XML_LRM_ATTR_INTERVAL;
-    value = g_hash_table_lookup(action->meta, field);
-    if (value != NULL) {
-        interval_ms = crm_parse_interval_spec(value);
-
-    } else if ((xml_obj == NULL) && !strcmp(action->task, RSC_STATUS)) {
-        /* An orphaned recurring monitor will not have any XML. However, we
-         * want the interval to be set, so the action can be properly detected
-         * as a recurring monitor. Parse it from the key in this case.
-         */
-        parse_op_key(action->uuid, NULL, NULL, &interval_ms);
-    }
     if (interval_ms > 0) {
         value_ms = crm_strdup_printf("%u", interval_ms);
         g_hash_table_replace(action->meta, strdup(field), value_ms);
 
-    } else if (value) {
+    } else if (g_hash_table_lookup(action->meta, field) != NULL) {
```

Comment:
g_hash_table_remove() will do nothing if nothing matches, so we don't really need a test here anymore

Reply:
Done.

Reply:
This line somehow ends up being important.  That's what was causing the scheduler tests to fail.

Reply:
What the heck? That scares me. I don't see how that could possibly matter.

I don't think pacemaker ever inserts hash table entries with a key and a NULL value (the only case where I think the behavior would be different), and even if it did, I would expect a nonexistent key and a NULL value for the key to be treated equivalently by the rest of the code anyway, because we only ever use g_hash_table_lookup() which returns NULL either way.

Reply:
We definitely do have keys where the value is NULL (or, we do now with my patches - I guess I'd need to test this against master too):

```
(pe_eval_expr)  trace: Expression Dummy-options-timeout-rule/rule-Dummy failed
(unpack_attr_set)       trace: Adding attributes from op_defaults-options
(populate_hash)         trace: Setting attribute: timeout = 5s
####### interval = (null)
```

The docs for g_hash_table_lookup have a warning for if the value is NULL.  See https://developer.gnome.org/glib/stable/glib-Hash-Tables.html#g-hash-table-lookup.

Reply:
Verified on master - an fprintf in the same-ish spot also returns keys with a value of NULL.

Reply:
If you're printing g_hash_table_lookup(), that's most likely a nonexistent key rather than an key that's present with a NULL value. Either way I don't think the rest of the code would treat those cases differently.

I'm at a loss. I keep staring it but I don't see how it can matter.